sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkDoubleCounter.java

```diff
@@ -109,7 +101,7 @@ public ObservableDoubleMeasurement buildObserver() {
     }
 
     @Override
-    public DoubleCounterAdviceConfigurer setAttributes(List<AttributeKey<?>> attributes) {
+    public ExtendedDoubleCounterBuilder setAttributesAdvice(List<AttributeKey<?>> attributes) {
```

Comment:
can we return this external interface without causing trouble for people using this class without the incubator code present?

Reply:
The incubator code is always present on the classpath. Its just an implementation dependency instead of api, so users have to add it as a dependency of their own to access the extended behavior. Including the incubator as a implementation dependency is acceptable because its not part of the public API of the SDK, and it doesn't include any transitive dependencies that would make it questionable.